### PR TITLE
empty list with spread operator is accepted

### LINKED ISSUE #6: List spread with no initial values should be parsed
The following should parse
```gleam
[..a]
```

### PR DESCRIPTION
**Pull request by CrowdHailer**
fixes #6

### CODE REVIEW

File: src/glance.gleam [hunk 1]
```diff
@@ -1471,7 +1471,18 @@ fn list(
 
     [#(t.Comma, _), #(t.RightSquare, _), ..tokens] if acc != [] ->
       Ok(#(list.reverse(acc), None, tokens))
+    [#(t.DotDot, _), #(t.RightSquare, _) as close, ..tokens] -> {
+      case discard {
+        None -> unexpected_error([close, ..tokens])
+        Some(discard) -> Ok(#(list.reverse(acc), Some(discard), tokens))
+      }
+    }
```
Comment (lpil) [inline src/glance.gleam:1479]: Is this used? I think the test is for the case below, I'm not sure what this one is for.

**Review by lpil — COMMENTED** [review]
Thank you. I've left one comment inline.

Would you mind updating the changelog also? Thank you

File: src/glance.gleam (hunk 1 above)
Comment (CrowdHailer) [reply src/glance.gleam:1479]: I added a test case for it.

**Comment by CrowdHailer** [comment]
Changelog entry added

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**